### PR TITLE
Fix #958: add more logging fields to retry job exceptions

### LINKED ISSUE #958: Add more context fields to queue exception logs
bug id, etc.

### PR DESCRIPTION
**Pull request by leplatrem**
Fix #958

### CODE REVIEW

File: jbi/retry.py [hunk 1]
```diff
@@ -52,8 +52,16 @@ async def retry_failed(item_executor=runner.execute_action, queue=get_dl_queue()
                     await queue.done(item)
                     metrics["events_processed"] += 1
                 except Exception:
-                    logger.exception("failed to reprocess event %s.", item.identifier)
                     metrics["events_failed"] += 1
+                    logger.exception(
+                        "failed to reprocess event %s.",
+                        item.identifier,
+                        extra={
+                            "metrics": metrics,
```
Comment (grahamalama) [inline jbi/retry.py:60]: I'm not sure that it's helpful to emit the metrics here. A dump of the item and the bug id make sense though.

File: jbi/retry.py [hunk 2]
```diff
@@ -67,8 +75,12 @@ async def retry_failed(item_executor=runner.execute_action, queue=get_dl_queue()
                         metrics["events_skipped"] += pending_events - 1
                         break
         except Exception:
-            logger.exception("failed to parse events for bug %d.", bug_id)
             metrics["bugs_failed"] += 1
+            logger.exception(
+                "failed to parse events for bug %d.",
+                bug_id,
+                extra={"metrics": metrics, "bug": {"id": bug_id}},
```
Comment (grahamalama) [inline jbi/retry.py:82]: Same here, with the questions about metrics being included in individual log lines

**Review by grahamalama — COMMENTED** [review]
I added some questions about metrics. Maybe instead of including the metrics in each log line, we can add like a "retry run id" (with a better name), and then include that in each log line. Then we can quickly find all of the logs for a particular retry run (with the metrics appearing on the last line as a summary for that run)

**Comment by alexcottner** [comment]
I'd like to see something added to `execute_or_queue` in `runner.py` as well. If `execute_action` throws an error, we should log an error with the event identifier so we can alert.

File: jbi/retry.py (hunk 1 above)
Comment (alexcottner) [reply jbi/retry.py:60]: I agree, I think logging the metrics multiple times could cause confusion. The rest of this looks good though.

**Review by grahamalama — APPROVED** [review]

**Review by alexcottner — APPROVED** [review]

**Pull request merged**